src/main/scala/com/twitter/finagle/websocket/WebSocketHandler.scala

```diff
@@ -10,11 +9,38 @@ import org.jboss.netty.handler.codec.http.websocketx._
 import org.jboss.netty.handler.codec.http.{HttpHeaders, HttpRequest, HttpResponse}
 import scala.collection.JavaConversions._
 
+private[finagle] class CancelledException extends Exception
+
 class WebSocketHandler extends SimpleChannelHandler {
   protected[this] val messagesBroker = new Broker[String]
   protected[this] val binaryMessagesBroker = new Broker[Array[Byte]]
   protected[this] val closer = new Promise[Unit]
 
+  implicit class RichChannelFuture(future: ChannelFuture) {
```

Comment:
Since this is only used in a couple of places, it might be a little cleaner to define conversion methods that are used directly.